OpenCL/m21920_a3-pure.cl

```diff
@@ -0,0 +1,274 @@
+/**
+ * Author......: See docs/credits.txt
+ * License.....: MIT
+ */
+
+#define NEW_SIMD_CODE
+
+#ifdef KERNEL_STATIC
+#include "inc_vendor.h"
+#include "inc_types.h"
+#include "inc_platform.cl"
+#include "inc_common.cl"
+#include "inc_simd.cl"
+#include "inc_hash_sha256.cl"
+#include "inc_hash_ripemd160.cl"
+#endif
+
+KERNEL_FQ void m21920_mxx (KERN_ATTR_VECTOR ())
+{
+  /**
+   * modifier
+   */
+
+  const u64 lid = get_local_id (0);
+  const u64 gid = get_global_id (0);
+
+  if (gid >= gid_max) return;
+
+  /**
+   * base
+   */
+
+  const u32 pw_len = pws[gid].pw_len;
+
+  u32x w[64] = { 0 };
+
+  for (u32 i = 0, idx = 0; i < pw_len; i += 4, idx += 1)
+  {
+    w[idx] = pws[gid].i[idx];
+  }
+
+  sha256_ctx_t ctx_base;
+
+  sha256_init (&ctx_base);
+
+  sha256_update_global_swap (&ctx_base, salt_bufs[salt_pos].salt_buf, salt_bufs[salt_pos].salt_len);
+
+  /**
+   * loop
+   */
+
+  u32x _w0[4];
+  u32x _w1[4];
+  u32x _w2[4];
+  u32x _w3[4];
+
+  u32x _v0[4];
+  u32x _v1[4];
+  u32x _v2[4];
+  u32x _v3[4];
+
+  u32x w0l = w[0];
+
+  for (u32 il_pos = 0; il_pos < il_cnt; il_pos += VECT_SIZE)
+  {
+    const u32x w0r = words_buf_r[il_pos / VECT_SIZE];
+
+    const u32x w0 = w0l | w0r;
+
+    w[0] = w0;
+
+    sha256_ctx_vector_t ctx_inner;
+
+    sha256_init_vector (&ctx_inner);
+
+    sha256_update_vector_swap (&ctx_inner, w, pw_len);
+
+    sha256_final_vector (&ctx_inner);
+
+    sha256_ctx_vector_t ctx_mid;
+
+    sha256_init_vector_from_scalar (&ctx_mid, &ctx_base);
+
+    _w0[0] = ctx_inner.h[0];
+    _w0[1] = ctx_inner.h[1];
+    _w0[2] = ctx_inner.h[2];
+    _w0[3] = ctx_inner.h[3];
+    _w1[0] = ctx_inner.h[4];
+    _w1[1] = ctx_inner.h[5];
+    _w1[2] = ctx_inner.h[6];
+    _w1[3] = ctx_inner.h[7];
+
+    _w2[0] = 0;
+    _w2[1] = 0;
+    _w2[2] = 0;
+    _w2[3] = 0;
+    _w3[0] = 0;
+    _w3[1] = 0;
+    _w3[2] = 0;
+    _w3[3] = 0;
+
+    sha256_update_vector_64 (&ctx_mid, _w0, _w1, _w2, _w3, 32);
+
+    sha256_final_vector (&ctx_mid);
+
+    _v0[0] = hc_swap32(ctx_mid.h[0]);
```

Comment:
please add spaces: `hc_swap32 (ctx_mid.h[0]);`

this is also true for the following lines and other hc_swap32 () etc function calls.